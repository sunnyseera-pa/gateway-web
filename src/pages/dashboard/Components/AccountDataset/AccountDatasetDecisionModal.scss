@import '../../../../css/variables';
@import '../../../../css/typography';

.account-dataset-decision-modal {
	box-sizing: border-box;

	.modal-dialog {
		max-width: 620px !important;
	}

	.rbt-input {
		font-size: 14px !important;
	}

	.rbt-close {
		color: #475da7 !important;
		margin-top: 0px !important;
	}
}

.Use-this-tool-to-fin {
	font-size: 14px;
	color: #3c3c3b;
}

.radio-buttons-container {
	display: flex;
	flex-direction: column;
}

.account-dataset-decision-title {
	display: flex;
	flex-direction: row;
	align-items: center;
}

.account-dataset-decision-footer {
	display: flex;
	width: 100%;

	border-top: 1px solid #e7e9ed;

	&--wrap {
		display: flex;
		justify-content: space-between;
		width: 100%;
		padding: 16px;
	}
}

.account-dataset-decision-header {
	display: flex;
	width: 100%;
	background: #ffffff;
	border-bottom: 1px solid #e7e9ed;

	&--wrap {
		width: 100%;
		padding: 32px 24px 32px 32px;
	}
}

.account-dataset-decision-body {
	display: flex;
	width: 100%;
	background: #ffffff;
	border-bottom: 1px solid #e7e9ed;
	min-height: 358px;

	&--wrap {
		width: 100%;
		padding: 24px 32px 32px 32px;
	}
}

.account-dataset-decision-modal-close {
	position: absolute;
	top: 16px;
	right: 16px;
	width: 17px;
	height: 17px;
	color: #111111;
}
