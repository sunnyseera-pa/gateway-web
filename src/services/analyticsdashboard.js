import { useQuery } from 'react-query';
import { apiURL } from '../configs/url.config';
import { getRequest } from '../utils/requests';

const getTotalUsers = options => {
	return getRequest(`${apiURL}/analyticsdashboard/totalusers`, options);
};

const getUsersPerMonth = options => {
	return getRequest(`${apiURL}/analyticsdashboard/userspermonth`, options);
};

const useGetTotalUsers = (requestOptions, queryOptions = { queryKey: 'getTotalUsers' }) => {
	return useQuery({
		...queryOptions,
		queryFn: () => getTotalUsers(requestOptions),
	});
};

const useGetUsersPerMonth = (requestOptions, queryOptions = { queryKey: 'getUsersPerMonth' }) => {
	return useQuery({
		...queryOptions,
		queryFn: () => getUsersPerMonth(requestOptions),
	});
};

export default {
	getTotalUsers,
	getUsersPerMonth,
	useGetTotalUsers,
	useGetUsersPerMonth,
};
