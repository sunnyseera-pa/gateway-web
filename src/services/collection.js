import { useMutation, useQuery } from 'react-query';
import { apiURL } from '../configs/url.config';
import { getRequest, postRequest } from '../utils/requests';

export const getCollectionRequest = (_id, options) => {
	return getRequest(`${apiURL}/collections/${_id}`, options);
};

export const getCollectionRelatedObjectsRequest = (_id, options) => {
	return getRequest(`${apiURL}/collections/relatedobjects/${_id}`, options);
};

export const postCollectionCounterUpdateRequest = (data, options) => {
	return postRequest(`${apiURL}/collectioncounter/update`, data, options);
};

const useGetCollectionRequest = (_id, requestOptions, queryOptions = { queryKey: 'getCollectionRequest' }) => {
	return useQuery({
		...queryOptions,
		queryKey: [queryOptions.queryKey, _id],
		queryFn: async ({ queryKey }) => getCollectionRequest(queryKey[1], requestOptions),
	});
};

const useGetCollectionRelatedObjectsRequest = (_id, requestOptions, queryOptions = { queryKey: 'getCollectionRelatedObjectsRequest' }) => {
	return useQuery({
		...queryOptions,
		queryKey: [queryOptions.queryKey, _id],
		queryFn: async ({ queryKey }) => getCollectionRelatedObjectsRequest(queryKey[1], requestOptions),
	});
};

const usePostCollectionCounterUpdateRequest = (requestOptions, mutateOptions = { queryKey: 'postCollectionCounterUpdateRequest' }) => {
	return useMutation(data => postCollectionCounterUpdateRequest(data, requestOptions), {
		mutateOptions,
	});
};

export default {
	getCollectionRequest,
	getCollectionRelatedObjectsRequest,
	postCollectionCounterUpdateRequest,
	useGetCollectionRequest,
	useGetCollectionRelatedObjectsRequest,
	usePostCollectionCounterUpdateRequest,
};
