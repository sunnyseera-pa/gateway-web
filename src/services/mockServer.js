import { setupServer } from 'msw/node';
import mswDatasets from './datasets/mockMsw';
import mswDatasetOnboarding from './dataset-onboarding/mockMsw';
import mswPostDatasetActivityLog from './activitylog/mockMsw';

const handlers = [...mswDatasets, ...mswDatasetOnboarding, ...mswPostDatasetActivityLog];

export const server = setupServer(...handlers);
