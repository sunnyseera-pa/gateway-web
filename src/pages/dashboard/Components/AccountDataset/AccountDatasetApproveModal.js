import { Formik } from 'formik';
import React, { Suspense } from 'react';
import { Button, Form, Modal } from 'react-bootstrap';
import { useTranslation } from 'react-i18next';
import * as Yup from 'yup';
import { STATUS_INREVIEW, TEXTAREA_ROWS } from '../../../../configs/constants';
import { ReactComponent as CloseButtonSvg } from '../../../../images/close-alt.svg';
import datasetOnboardingService from '../../../../services/dataset-onboarding/dataset-onboarding';
import './AccountDatasetDecisionModal.scss';

const AccountDatasetApproveModal = ({ id, open, closed, goToNext, showGoToNext, handleApprove }) => {
	const { t } = useTranslation();

	const approveDataset = React.useCallback(
		async values => {
			const payload = {
				...values,
				id,
				applicationStatus: 'approved',
			};

			await datasetOnboardingService.putDatasetOnboarding(id, payload);
		},
		[id]
	);

	const handleSubmit = React.useCallback(
		async values => {
			await approveDataset(values);

			handleApprove({
				publisher: '',
				tab: STATUS_INREVIEW,
				message: `You have approved the dataset`,
			});
		},
		[id]
	);

	return (
		<Suspense fallback={t('loading')}>
			<Modal show={open} onHide={closed} className='decisionModal' size='lg' aria-labelledby='contained-modal-title-vcenter' centered>
				<div className='decisionModal-header'>
					<div className='decisionModal-header--wrap'>
						<div className='decisionModal-head'>
							<h1 className='black-20-semibold'>{t('dataset.approvalModal.title')}</h1>
							<CloseButtonSvg className='decisionModal-head--close' onClick={closed} />
						</div>
					</div>
				</div>
				<Formik
					initialValues={{
						applicationStatusDesc: '',
					}}
					validationSchema={Yup.object({
						applicationStatusDesc: Yup.string().max(1500, 'Description must be less than 1500 characters'),
					})}
					onSubmit={handleSubmit}>
					{({ values, errors, isValid, handleChange, handleBlur, handleSubmit }) => (
						<Form onSubmit={handleSubmit} onBlur={handleBlur}>
							<div className='decisionModal-body'>
								<div className='decisionModal-body--wrap'>
									<p data-testid='description'>{t('dataset.approvalModal.description')}</p>
									<Form.Group>
										<label for='applicationStatusDesc' className='black-14'>
											{t('dataset.approvalModal.applicationStatus')}
											<span>{values.applicationStatusDesc.length}/1500</span>
										</label>
										<Form.Control
											as='textarea'
											data-testid='dataset-reject-applicationStatusDesc'
											id='applicationStatusDesc'
											name='applicationStatusDesc'
											type='text'
											onChange={handleChange}
											value={values.applicationStatusDesc}
											onBlur={handleBlur}
											rows={TEXTAREA_ROWS}
										/>
										{errors.applicationStatusDesc ? <div className='errorMessages'>{errors.applicationStatusDesc}</div> : null}
									</Form.Group>
								</div>
							</div>
							<div className='decisionModal-footer'>
								<div data-testid='button-container' className='decisionModal-footer--wrap'>
									<Button
										className='button-secondary'
										onClick={() => {
											closed();
										}}>
										{t('dataset.approvalModal.buttons.cancel')}
									</Button>
									<Button
										disabled={!isValid}
										type='submit'
										data-testid='approve-button'
										className='button-secondary'
										style={{ marginLeft: '10px' }}>
										{t('dataset.approvalModal.buttons.approve')}
									</Button>
									<Button
										disabled={!showGoToNext || !isValid}
										className='button-secondary'
										style={{ marginLeft: '10px' }}
										onClick={async () => {
											await approveDataset(values);

											goToNext();
										}}>
										{t('dataset.approvalModal.buttons.approveAndGoToNext')}
									</Button>
								</div>
							</div>
						</Form>
					)}
				</Formik>
			</Modal>
		</Suspense>
	);
};

export default AccountDatasetApproveModal;
