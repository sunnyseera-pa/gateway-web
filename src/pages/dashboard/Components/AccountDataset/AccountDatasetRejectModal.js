import React from 'react';
import { Modal } from 'react-bootstrap';
import { ReactComponent as CloseButtonSvg } from '../../../../images/close-alt.svg';
import './AccountDatasetDecisionModal.scss';
import _ from 'lodash';

const AccountDatasetRejectModal = ({
	open,
	closed,
    onReject,
	onRejectAndGoToNext
}) => {
	return (
		<Modal
			show={open}
			onHide={closed}
			className='account-dataset-decision-modal'
			size='lg'
			aria-labelledby='contained-modal-title-vcenter'
			centered>
			<div className='account-dataset-decision-header'>
				<CloseButtonSvg className='account-dataset-decision-modal-close' onClick={closed} />
				<div className='account-dataset-decision-header--wrap'>
					<h4 className='black-20'>Reject this version of this dataset metadata</h4>
				</div>
			</div>

			<div className='account-dataset-decision-body'>
				<div className='account-dataset-decision-body--wrap'>Placeholder content for now</div>
			</div>
			<div className='account-dataset-decision-footer'>
				<div className='account-dataset-decision-footer--wrap'>
					<button
						className='button-secondary'
						style={{ marginLeft: 'auto' }}
						onClick={() => {
							closed();
						}}>
						No, nevermind
					</button>
					<button
						className='button-secondary'
						style={{ marginLeft: 'auto' }}
						onClick={() => {
							onReject();
							closed();
						}}>
						Reject
					</button>
					<button
						className='button-secondary'
						style={{ marginLeft: 'auto' }}
						onClick={() => {
							onRejectAndGoToNext();
							closed();
						}}>
						Reject and go to next
					</button>
				</div>
			</div>
		</Modal>
	);
};

export default AccountDatasetRejectModal;
