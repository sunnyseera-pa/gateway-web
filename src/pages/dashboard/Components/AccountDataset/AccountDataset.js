import _ from 'lodash';
import React, { useState } from 'react';
import { Button, OverlayTrigger, Popover } from 'react-bootstrap';
import { useParams } from 'react-router';
import { useAuth } from '../../../../context/AuthContext';
import serviceDatasetOnboarding from '../../../../services/dataset-onboarding/dataset-onboarding';
import serviceDatasets from '../../../../services/datasets/datasets';
import { getTeam } from '../../../../utils/auth';
import utils from '../../../../utils/DataSetHelper.util';
import ActionBar from '../../../commonComponents/actionbar/ActionBar';
import DatasetCard from '../../../commonComponents/DatasetCard';
import Loading from '../../../commonComponents/Loading';
import NotFound from '../../../commonComponents/NotFound';
import AccountContent from '../AccountContent';
import AccountDatasetApproveModal from './AccountDatasetApproveModal';
import AccountDatasetRejectModal from './AccountDatasetRejectModal';

const AccountDataset = props => {
	const { id } = useParams();
	const { userState } = useAuth();
	const [team, setTeam] = useState();
	const [state, setState] = useState({
		showPrevious: true,
		showDisabled: true,
		statusError: false,
		showApproveDatasetModal: false,
		showRejectDatasetModal: false
	});

	const dataDataset = serviceDatasets.useGetDataset(id);
	const publisherId = utils.getPublisherID(userState[0], team);
	const dataPublisher = serviceDatasetOnboarding.useGetPublisher(Array.isArray(publisherId) ? publisherId[0] : publisherId);

	React.useEffect(() => {
		setTeam(getTeam(props));
	}, [id]);

	const goToDataset = i => {
		if (dataPublisher.data) {
			const {
				data: {
					data: { listOfDatasets: datasets },
				},
			} = dataPublisher.data;

			const inReview = datasets.filter(dataset => dataset.activeflag === props.activeflag);

			const currentIndex = _.findIndex(inReview, dataset => {
				return dataset.pid == id;
			});

			setState({
				showNext: currentIndex < inReview.length - 1,
				showPrevious: currentIndex > 0,
			});

			if (currentIndex === -1) {
				setState({
					statusError: true,
				});

				return;
			}

			return inReview[currentIndex + i].pid;
		}
	};

	React.useEffect(() => {
		goToDataset(0);
	}, [dataPublisher.data]);

	const handlePaginationClick = React.useCallback(
		i => {
			const pid = goToDataset(i);

			window.location.assign(`/account/datasets/${pid}`);
		},
		[id, dataPublisher.data, team]
	);

	const goToNext = () => {
		if (showNext) {
			handlePaginationClick(1);
		}
	};

	const { showPrevious, showNext, statusError, showRejectDatasetModal, showApproveDatasetModal } = state;

	if (!dataDataset.data && dataDataset.isError && dataDataset.isFetched) {
		return (
			<AccountContent>
				<NotFound word='dataset' />
			</AccountContent>
		);
	}

	if (statusError) {
		return (
			<AccountContent>
				<NotFound text='The activity log for this dataset cannot be accessed. It must be set to in review.' />
			</AccountContent>
		);
	}

	if (dataDataset.isLoading || dataPublisher.isLoading) {
		return (
			<AccountContent>
				<Loading />
			</AccountContent>
		);
	}

	const {
		data: { data: dataset },
	} = dataDataset.data;

	return (
		<AccountContent>
			<DatasetCard
				id={dataset._id}
				title={dataset.name}
				publisher={dataset.datasetv2.summary.publisher.name}
				version={dataset.datasetVersion}
				isDraft={true}
				datasetStatus={dataset.activeflag}
				timeStamps={dataset.timestamps}
				completion={dataset.percentageCompleted}
				listOfVersions={dataset.listOfVersions || []}
			/>
			<ActionBar userState={userState}>
				<div className='action-bar-actions'>
					{showPrevious && !statusError && (
						<Button variant='light' onClick={() => handlePaginationClick(-1)}>
							Previous
						</Button>
					)}
					{showNext && !statusError && (
						<Button variant='light' onClick={() => handlePaginationClick(1)}>
							Next
						</Button>
					)}
					<OverlayTrigger 
						trigger='click'
						key='top' 
						placement='top'
						overlay={
							<Popover id='make-a-decision-popover'>
								<Popover.Title as h3>
									<span data-testid='make-as-decision-overlay'>Make a decision</span>
								</Popover.Title>
								<Popover.Content>
									<Button variant="link" onClick={() =>
											setState({ showApproveDatasetModal: true })}>Approve</Button>
									<Button variant="link" onClick={() =>
											setState({ showRejectDatasetModal: true })}>Reject</Button>
								</Popover.Content>
							</Popover>
						}
					>
						<Button variant='outline-secondary'>
							Make a decision
						</Button>
					</OverlayTrigger>
				</div>
			</ActionBar>
			<AccountDatasetApproveModal
				id={dataset._id}
				open={showApproveDatasetModal}
				closed={() => setState({ showApproveDatasetModal: false })}
				goToNext={() => goToNext()}
				showGoToNext={showNext} />
			<AccountDatasetRejectModal
				id={dataset._id}
				open={showRejectDatasetModal}
				closed={() => setState({ showRejectDatasetModal: false })}
				goToNext={() => goToNext()}
				showGoToNext={showNext} />
		</AccountContent>
	);
};

AccountDataset.defaultProps = {
	activeflag: 'inReview',
};

export default AccountDataset;
