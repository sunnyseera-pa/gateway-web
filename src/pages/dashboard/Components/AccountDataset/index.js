import AccountDataset from './AccountDataset';

export default AccountDataset;
