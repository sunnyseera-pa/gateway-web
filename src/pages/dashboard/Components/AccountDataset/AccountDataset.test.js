import { render, waitFor } from '@testing-library/react';
import React from 'react';
import reactRouter from 'react-router-dom';
import AccountDataset from '.';
import { server } from '../../../../services/mockServer';

jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	Redirect: props => {
		mockRedirect(props);
		return null;
	},
}));

jest.mock('../../../commonComponents/DatasetCard', () => props => {
	mockDatasetCard(props);
	return <div />;
});

jest.mock('../ActivityLogCard', () => props => {
	mockActivityLogCard(props);
	return <div />;
});

const mockDatasetCard = jest.fn();
const mockActivityLogCard = jest.fn();
const mockPush = jest.fn();
const mockRedirect = jest.fn();

const props = {
	location: {},
};

let wrapper;

describe('Given the AccountDataset component', () => {
	beforeAll(() => {
		server.listen();

		jest.spyOn(reactRouter, 'useHistory').mockImplementation(() => ({
			push: mockPush,
		}));
	});

	afterEach(() => {
		server.resetHandlers();
	});

	afterAll(() => {
		server.close();
	});

	describe('When it is rendered', () => {
		beforeAll(() => {
			jest.spyOn(reactRouter, 'useParams').mockReturnValue({
				id: 'd5c99a71-c039-4a0b-9171-dba8a1c33154',
			});

			wrapper = render(<AccountDataset {...props} />, {
				wrapper: Providers,
			});
		});

		it('Then matches the previous snapshot', async () => {
			expect(wrapper.container).toMatchSnapshot();
		});

		it('Then shows a loader', async () => {
			await waitFor(() => expect(wrapper.getByText('Loading...')).toBeTruthy());
		});

		it('Then calls the ActivityLogCard the correct number of times', async () => {
			await waitFor(() => expect(mockActivityLogCard).toHaveBeenCalledTimes(4));
		});

		describe('And the view form is clicked', () => {
			beforeAll(async () => {
				await waitFor(() => expect(wrapper.getByText('View form')).toBeTruthy());

				const button = wrapper.getByText('View form');

				await fireEvent.click(button);
			});

			afterAll(() => {
				mockPush.mockReset();
			});

			it('Then loads the new dataset', () => {
				expect(mockPush).toHaveBeenCalledWith(`/dataset-onboarding/618128a2c312ff5648a20850`);
			});
		});

		describe('And the next button is clicked', () => {
			beforeAll(async () => {
				await waitFor(() => expect(wrapper.queryAllByText('Next')).toBeTruthy());

				const button = wrapper.queryAllByText('Next')[0];

				await fireEvent.click(button);
			});

			afterAll(() => {
				mockPush.mockReset();
			});

			it('Then loads the new dataset', () => {
				expect(mockPush).toHaveBeenCalledWith('/account/datasets/4932179f-1c9c-40a0-81b5-9b499aff7a64');
			});
		});

		describe('And the Make a decision button is clicked', () => {
			beforeAll(async () => {
				await waitFor(() => expect(wrapper.getByText('Make a decision')).toBeTruthy());

				const button = wrapper.getByText('Make a decision');

				await fireEvent.click(button);
			});

			it('Then loads the Make a decision overlay', () => {
				expect(wrapper.container).toMatchSnapshot();
			});

			describe('And the Reject button is clicked', () => {
				beforeAll(async () => {
					await waitFor(() => expect(wrapper.getByText('Reject')).toBeTruthy());

					const button = wrapper.queryAllByText('Reject')[0];

					await fireEvent.click(button);
				});

				afterAll(() => {
					const closeModalButton = wrapper.queryByTestId('close-modal');
					if (closeModalButton) {
						fireEvent.click(closeModalButton);
					}
				});

				it('Then loads the Reject dataset modal', () => {
					const { queryByText } = wrapper;
					expect(queryByText('Reject this version of this dataset metadata')).toBeTruthy();
				});

				it('Then does not load the Approve dataset modal', () => {
					const { queryByText } = wrapper;
					expect(queryByText('Approve this version of this dataset metadata')).toBeFalsy();
				});
			});

			describe('And the Approve button is clicked', () => {
				beforeAll(async () => {
					await waitFor(() => expect(wrapper.getByText('Approve')).toBeTruthy());

					const button = wrapper.queryAllByText('Approve')[0];

					await fireEvent.click(button);
				});

				afterAll(() => {
					const closeModalButton = wrapper.queryByTestId('close-modal');
					if (closeModalButton) {
						fireEvent.click(closeModalButton);
					}
				});

				it('Then loads the Approve dataset modal', () => {
					const { queryByText } = wrapper;
					expect(queryByText('Approve this version of this dataset metadata')).toBeTruthy();
				});

				it('Then does not load the Reject dataset modal', () => {
					const { queryByText } = wrapper;
					expect(queryByText('Reject this version of this dataset metadata')).toBeFalsy();
				});
			});
		});

		describe('And the previous button is clicked', () => {
			beforeAll(async () => {
				await waitFor(() => expect(wrapper.queryAllByText('Previous')).toBeTruthy());

				const button = wrapper.queryAllByText('Previous')[0];

				await fireEvent.click(button);
			});

			afterAll(() => {
				mockPush.mockReset();
			});

			it('Then loads the new dataset', () => {
				expect(mockPush).toHaveBeenCalledWith('/account/datasets/0a048419-0796-46fb-ad7d-91e650a6c742');
			});
		});
	});

	describe('When the dataset is not in review', () => {
		beforeAll(async () => {
			jest.spyOn(reactRouter, 'useParams').mockReturnValue({
				id: '1f509fe7-e94f-48fe-af6a-81f2bf8a5270',
			});

			afterAll(() => {
				mockPush.mockReset();
			});

			it('Then loads the new dataset', () => {
				expect(mockPush).toHaveBeenCalledWith('/account?tab=datasets');
			});
		});

		it('Then loads the new dataset', async () => {
			await waitFor(() => expect(mockRedirect).toHaveBeenCalled());

			expect(mockRedirect).toHaveBeenCalledWith({
				to: '/account?tab=datasets',
			});
		});
	});

	describe('When the dataset is not valid', () => {
		beforeAll(async () => {
			jest.spyOn(reactRouter, 'useParams').mockReturnValue({
				id: 'invalid',
			});
		});

		afterAll(() => {
			mockPush.mockReset();
		});

		it('Then loads the new dataset', async () => {
			await waitFor(() => expect(mockRedirect).toHaveBeenCalled());

			expect(mockRedirect).toHaveBeenCalledWith({
				to: '/account?tab=datasets',
			});
		});
	});
});
