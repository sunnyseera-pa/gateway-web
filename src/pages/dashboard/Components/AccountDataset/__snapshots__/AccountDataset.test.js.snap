// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the AccountDataset component When it is rendered Then has the correct number of versions 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="col-1"
    />
    <div
      class="col-10"
    >
      <div />
      <div
        class="row"
      >
        <div
          class="col-md-12"
        >
          <div
            class="layoutCard mb-0"
          >
            <div
              class="datasetHeader mb-0 mt-2"
            >
              <div
                class="datasetHeader-title"
              >
                <h1>
                  Version 4.0.0
                </h1>
              </div>
              <div
                class="datasetHeader-status"
              >
                <div
                  class="sla sla-red"
                >
                  Rejected
                </div>
              </div>
            </div>
            <div
              class="body"
            >
              Text goes here
            </div>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-12"
        >
          <div
            class="layoutCard mb-0"
          >
            <div
              class="datasetHeader mb-0 mt-2"
            >
              <div
                class="datasetHeader-title"
              >
                <h1>
                  Version 3.0.0
                </h1>
              </div>
              <div
                class="datasetHeader-status"
              >
                <div
                  class="sla sla-gray"
                >
                  Archived
                </div>
              </div>
            </div>
            <div
              class="body"
            >
              Text goes here
            </div>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-12"
        >
          <div
            class="layoutCard mb-0"
          >
            <div
              class="datasetHeader mb-0 mt-2"
            >
              <div
                class="datasetHeader-title"
              >
                <h1>
                  Version 2.0.0
                </h1>
              </div>
              <div
                class="datasetHeader-status"
              >
                <div
                  class="sla sla-red"
                >
                  Rejected
                </div>
              </div>
            </div>
            <div
              class="body"
            >
              Text goes here
            </div>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-12"
        >
          <div
            class="layoutCard mb-0"
          >
            <div
              class="datasetHeader mb-0 mt-2"
            >
              <div
                class="datasetHeader-title"
              >
                <h1>
                  Version 1.0.0
                </h1>
              </div>
              <div
                class="datasetHeader-status"
              >
                <div
                  class="sla sla-gray"
                >
                  Archived
                </div>
              </div>
            </div>
            <div
              class="body"
            >
              Text goes here
            </div>
          </div>
        </div>
      </div>
      <div
        class="actionBar"
      >
        <div
          class="action-bar-actions"
        >
          <button
            class="btn btn-light"
            type="button"
          >
            Next
          </button>
        </div>
      </div>
    </div>
    <div
      class="col-1"
    />
  </div>
</div>
`;

exports[`Given the AccountDataset component When it is rendered Then matches the previous snapshot 1`] = `
<div>
  <div
    class="row"
  >
    <div
      class="col-1"
    />
    <div
      class="col-10"
    >
      <div />
      <div
        class="row"
      >
        <div
          class="col-md-12"
        >
          <div
            class="layoutCard mb-0"
          >
            <div
              class="datasetHeader mb-0 mt-2"
            >
              <div
                class="datasetHeader-title"
              >
                <h1>
                  Version 4.0.0
                </h1>
              </div>
              <div
                class="datasetHeader-status"
              >
                <div
                  class="sla sla-red"
                >
                  Rejected
                </div>
              </div>
            </div>
            <div
              class="body"
            >
              Text goes here
            </div>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-12"
        >
          <div
            class="layoutCard mb-0"
          >
            <div
              class="datasetHeader mb-0 mt-2"
            >
              <div
                class="datasetHeader-title"
              >
                <h1>
                  Version 3.0.0
                </h1>
              </div>
              <div
                class="datasetHeader-status"
              >
                <div
                  class="sla sla-gray"
                >
                  Archived
                </div>
              </div>
            </div>
            <div
              class="body"
            >
              Text goes here
            </div>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-12"
        >
          <div
            class="layoutCard mb-0"
          >
            <div
              class="datasetHeader mb-0 mt-2"
            >
              <div
                class="datasetHeader-title"
              >
                <h1>
                  Version 2.0.0
                </h1>
              </div>
              <div
                class="datasetHeader-status"
              >
                <div
                  class="sla sla-red"
                >
                  Rejected
                </div>
              </div>
            </div>
            <div
              class="body"
            >
              Text goes here
            </div>
          </div>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-md-12"
        >
          <div
            class="layoutCard mb-0"
          >
            <div
              class="datasetHeader mb-0 mt-2"
            >
              <div
                class="datasetHeader-title"
              >
                <h1>
                  Version 1.0.0
                </h1>
              </div>
              <div
                class="datasetHeader-status"
              >
                <div
                  class="sla sla-gray"
                >
                  Archived
                </div>
              </div>
            </div>
            <div
              class="body"
            >
              Text goes here
            </div>
          </div>
        </div>
      </div>
      <div
        class="actionBar"
      >
        <div
          class="action-bar-actions"
        >
          <button
            class="btn btn-light"
            type="button"
          >
            Next
          </button>
        </div>
      </div>
    </div>
    <div
      class="col-1"
    />
  </div>
</div>
`;
